Homework4/pupils.py

```diff
@@ -0,0 +1,18 @@
+# Каждый из N школьников некоторой школы знает Mi языков.
+# Определите, какие языки знают все школьники и языки,
+# которые знает хотя бы один из школьников.
+
+f = open('pupils.txt')
+
+dct_pupils = {}
+for i in range(int(f.readline())):
+    for j in range(int(f.readline())):
+        dct_pupils[i + 1] = dct_pupils.get(i + 1, set())
+        dct_pupils[i + 1].add(f.readline().strip())
+
+big_set = small_set = set()
```

Comment:
it's pretty unclear as you make big and small sets the same test and you still change big set and reassign small set.